lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,179 @@ def __init__(self, config):
+        if (bool(ws_id) == bool(ws_name)):  # xnor
+            raise ValueError(f"Exactly one of a '{_WSID}' or a '{_WSNAME}' parameter must be provided")
+        if not ws_id:
+            print(f"Translating workspace name {ws_name} to a workspace ID. Prefer submitting "
+                  + "a workspace ID over a mutable workspace name that may cause race conditions")
+            ws_id = self.dfu.ws_name_to_id(ws_name)
+        mass_params = {_WSID: ws_id, _INPUTS: [inputs]}
+        return mass_params
+
+    def _validate_mass_params(self, params):
+        ws_id = self._get_int(params.get(_WSID), _WSID)
+        if not ws_id:
+            raise ValueError(f"{_WSID} is required")
+        inputs = params.get(_INPUTS)
+        if not inputs or type(inputs) != list:
+            raise ValueError(f"{_INPUTS} field is required and must be a non-empty list")
+        for i, inp in enumerate(inputs, start=1):
+            if type(inp) != dict:
+                raise ValueError(f"Entry #{i} in {_INPUTS} field is not a mapping as required")
+            self._validate_params(inp)
+
+    def _get_int(self, putative_int, name, minimum=1):
+        if putative_int is not None:
+            if type(putative_int) != int:
+                raise ValueError(f"{name} must be an integer, got: {putative_int}")
+            if putative_int < minimum:
+                raise ValueError(f"{name} must be an integer >= {minimum}")
+        return putative_int
+
+    def _import_genbank_mass(self, params):
+
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        genome_names = []
+        genome_data = []
+        genome_meta = []
+
+        genome_objs = []
+        consolidated_files = []
+        input_directories = []
+
+        for idx, input_params in enumerate(inputs):
+
+            genome_obj = _Genome(self.cfg)
+            genome_names.append(input_params['genome_name'])
+
+            # construct the input directory staging area
+            input_directory = self._stage_input(input_params)
+            input_directories.append(input_directory)
+
+            # update default params
+            input_params = {**genome_obj.default_params, **input_params}
+            genome_meta.append(input_params['metadata'])
+            inputs[idx] = input_params
+
+            genome_obj.generate_parents = input_params.get('generate_missing_genes')
+            genome_obj.generate_ids = input_params.get('generate_ids_if_needed')
+            if input_params.get('genetic_code'):
+                genome_obj.code_table = input_params['genetic_code']
+            genome_objs.append(genome_obj)
+
+            # do the upload
+            files = self._find_input_files(input_directory)
+            consolidated_file = self._join_files_skip_empty_lines(files)
+            consolidated_files.append(consolidated_file)
+
+        # save files to shock mass
+        file_handles = self._save_files_to_blobstore(consolidated_files)
```

Comment:
I would parse the genomes before saving anything in case there's an error in the genome files. That way we fail before we save a ton of data

Reply:
so this save file to shock mass step is independent? I kind basically put it after upload assembly/genome step?

Reply:
You do

Reply:
when you say parse genomes, what exactly function are you referring to? _validate_existing_assembly?

Reply:
No idea. There must be some place where the genbank file is parsed and turned into an object. I'm saying do that before saving anything to the data stores so errors are caught up front

Reply:
👍

Reply:
It looks to me like the genbank files are still being parsed after the assemblies are saved

Reply:
Will fix in the next PR

Reply:
:+1: